Python/optimizer_analysis.c

```diff
@@ -1020,6 +990,14 @@ uop_abstract_interpret_single_inst(
             break;
         }
 
+        case PUSH_NULL: {
+            STACK_GROW(1);
+            _Py_UOpsSymbolicExpression *null_sym =  _Py_UOpsSymbolicExpression_New(ctx, PUSH_NULL, 0, NULL, 0);
```

Comment:
This won't be a singleton, because each time the originating_store will be different.

You should create a helper method to create singletons (ideally all their originating stores will be 0 or something).

Reply:
Okie I've implemented the Singletons. Note this requires `PUSH_NULL` to be labelled as "immutable"